### PR TITLE
Kill test process if it takes more than 5 minutes to execute

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

During fault injection, set a watchdog timer in PS to kill the process if it takes more than 5 minutes to execute.

## Testing

CI/CD

## Documentation

No.

### CODE REVIEW

File: scripts/Test-FaultInjection.ps1 [hunk 1]
```diff
@@ -22,6 +22,16 @@ Set-Content -Path ($TestProgram + ".fault.log") ""
 
 $iteration = 0
 
+$timer = New-Object System.Timers.Timer
+# Set timer for 5 minutes
+$timer.Interval = 300000
```
Comment (shpalani) [inline scripts/Test-FaultInjection.ps1:27]: Is this time value even for fault_injection_full?

Comment (Alan-Jowett) [reply scripts/Test-FaultInjection.ps1:27]: It shouldn't. Each iteration should take less than 5 minutes unless it's stuck.

File: scripts/Test-FaultInjection.ps1 [hunk 2]
```diff
@@ -22,6 +22,16 @@ Set-Content -Path ($TestProgram + ".fault.log") ""
...
+
+# When the watchdog timer fires, kill the test binary.
+Register-ObjectEvent -InputObject $timer -EventName Elapsed -Action {
+    Write-Error "Test binary exceede time"
```
Comment (shpalani) [inline scripts/Test-FaultInjection.ps1:31]: Correct spelling: exceeded.

Also, will the user know which binary exceeded the time to cancel?

File: scripts/Test-FaultInjection.ps1 (hunk 1 above)
Comment (shpalani) [reply scripts/Test-FaultInjection.ps1:27]: Thanks for clarifying. The reason of ask is, we have more APIs added with fault injection.

**Review by saxena-anurag — APPROVED** [review]

File: scripts/Test-FaultInjection.ps1 (hunk 2 above)
Comment (dthaler) [inline scripts/Test-FaultInjection.ps1:31]:
```suggestion
    Write-Error "Test binary exceeded time"
```

Comment (Alan-Jowett) [reply scripts/Test-FaultInjection.ps1]: This should gather a memory dump of the hung process, but that requires some additional work. The first version is to improve the reliability of the CI/CD pipeline.

Comment (shpalani) [reply scripts/Test-FaultInjection.ps1]: Can we have an issue tracking the improvement for the next step?

**Review by dthaler — APPROVED** [review]

File: scripts/Test-FaultInjection.ps1 [hunk 3]
```diff
@@ -22,6 +22,16 @@ Set-Content -Path ($TestProgram + ".fault.log") ""
 
 $iteration = 0
 
+$timer = New-Object System.Timers.Timer
+# Set timer for 5 minutes
+$timer.Interval = 300000
+
+# When the watchdog timer fires, kill the test binary.
```
Comment (dthaler) [inline scripts/Test-FaultInjection.ps1:29]: And will this be treated as a pass or a failure?

**Review by dthaler — APPROVED** [review]

**Pull request merged**